### PR TITLE
Add Proxy object support for several builtin operations

### PR DESCRIPTION
**Pull request by rerobika**
Related methods:
 - Array.isArray
 - JSON.stringify
 - JSON.parse
 - Object.prototype.toString

Also fixed some errors in the Promise builtin rotuines to enable more tests.

Signed-off-by: Robert Fancsik <frobert@inf.u-szeged.hu>

### CODE REVIEW

**Comment by rerobika** [comment]
Depends on https://github.com/Samsung/js_vendor_tc/pull/17.

File: src/runtime/Object.cpp [hunk 1]
```diff
@@ -1430,6 +1434,33 @@ IteratorObject* Object::entries(ExecutionState& state)
     return new ArrayIteratorObject(state, this, ArrayIteratorObject::TypeKeyValue);
 }
 
+ObjectPropertyNameVector Object::enumerableOwnNames(ExecutionState& state, Object* object)
```
[the remark this answers is not in the document]
Comment (rerobika) [reply src/runtime/Object.cpp]: Sure, I'll rename it.

[the remark this answers is not in the document]
Comment (bbrto21) [reply src/runtime/Object.cpp]: I mean that 'enumerableOwnProperties' is already implemented in Object.h.

[the remark this answers is not in the document]
Comment (rerobika) [reply src/runtime/Object.cpp]: Thank for the suggestion I've updated the PR.

**Pull request merged**